### ISSUE TITLE
Auto-email: Unable to send an Auto-email email even if Target Kick has already passed.

### ISSUE THREAD

**Issue by masarapmabuhay** [usbong/UsbongStore#64]
Hi Zent,
I hope you are doing well.

I encountered a bug while I was attempting to create a new batch to kick using the Auto-email component.

Even if the Target Kick datetime stamp has already passed, e.g. **2018-02-16 16:00:39**, and it's already **2018-02-16 16:10:39**, the Auto-email Admin page is saying that there's nothing to kick.

Would you be able to verify if this bug occurs on your local version as well?

Thank you.

Peace,
Mike

**Comment by zent-lim** [comment]
Good Day Mike,

(1) Could you check if it is the same issue as this one?
https://github.com/usbong/UsbongStore/issues/28

This happened before and we concluded that your mysql clock and your php
clock are not in sync.

(2) On the other hand, if symptoms persist please send me your sql file so
I can recreate the scenario on
my end.

Have a good day ahead!

**Comment by masarapmabuhay** [comment]
Hi Zent,
Thank you for the reply.

(1) No, the present datatime stamp is correct.

(2) OK. I have emailed you the newest SQL Commands of the Usbong Store.

Please let me 
know if you need
anything else.
 
Thank you.

Peace,
Mike

**Comment by zent-lim** [comment]
Hi Mike,

Thank you for sending me your sql.

Are you referring to this email and this queue?
Email: Need help with translation?
Queue: 34 (2018-02-16 16:00:39)

If yes, it's caused the queue was paused.

![paused2](https://user-images.githubusercontent.com/8021327/36336342-da5759ba-13c1-11e8-8703-34efd0b401bd.png)

I unpaused it, and things look fine:

![paused3](https://user-images.githubusercontent.com/8021327/36336386-4306be38-13c2-11e8-92b1-974e32237c53.png)

To unpause:

(1) Visit the queue url:
https://store.usbong.ph/index.php/auto-email/queue/18/1

(2) Click the play button

Have a good day ahead! :)

**Comment by masarapmabuhay** [comment]
Hi Zent,
Thank you for the reply.

No, I am not referring
to the email in the queue
that you mentioned.

May I kindly ask if you would
be able to create a new batch
and immediately send it?

In my case, doing so
would result to the issue 
in this thread.

Thank you.

Peace,
Mike

**Comment by zent-lim** [comment]
Hi MIke,

Copy.

I will investigate further.

**Comment by zent-lim** [comment]
Hi Mike,

I could not recreate your scenario.

I did the following:

(1) Create New Email

(2) Create New Queue in Email

(3) Go back to auto email home page

The new queue appeared correctly as shown below:

![test](https://user-images.githubusercontent.com/8021327/36339161-f112e984-13f9-11e8-9265-9e1be969cf63.png)

Could you give me more information on how to recreate the problem?

Have a good day ahead!

**Comment by masarapmabuhay** [comment]
Hi Zent,
Thank you for the replies.

I may be using an older version of the code on prod.
Would you be able to verify if our GitHub repo is updated?

Thank you.

Peace,
Mike
p.s. To replicate the issue, simply create a new batch using the "email_frame_single_template", which appears to be what you have already done.

**Comment by zent-lim** [comment]
Evening Mike,

Copy. I ll check master repo this evening.

Yeah. I already recreated the steps u mentioned.

Does ur prod have git installed?

Happy sunday!


On Feb 18, 2018 10:19 AM, "Masarap Mabuhay" <notifications@github.com>
wrote:

Hi Zent,
Thank you for the replies.

I may be using an older version of the code on prod.
Would you be able to verify if our GitHub repo is updated?

Thank you.

Peace,
Mike
p.s. To replicate the issue, simply create a new batch using the
"email_frame_single_template", which appears to be what you have already
done.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/usbong/UsbongStore/issues/64#issuecomment-366487279>,
or mute the thread

**Comment by masarapmabuhay** [comment]
Hi Zent,
Thank you for the reply.

Yes, Git is installed on prod.

Peace,
Mike

**Comment by zent-lim** [comment]
Pls do the ff instead.

This check is faster.

(1) using ssh go to UsbongStore/usbong_store

(2) run command below and pls send me result

git log -1

Note: this command tells us what version is in prod

(3) run the command below and pls send me result

git status

Note: this command identifies which files were changed w respect to the
code version u are using

After we get results from the commands i ll ask to do git diff on specific
files nxt.

Thanks.

**Comment by masarapmabuhay** [comment]
Hi Zent,
Thank you for the reply.

Git is installed on prod, but I do not use it as you likely do.

I have a downloaded version of the Git repo using SourceTree on my local machine, and then usually copy the "Model", "View", "Controller" to prod.

With this information, would you be able to tell me which files I should also make sure are updated on prod?

Thank you.

Peace,
Mike

**Comment by zent-lim** [comment]
Hi Mike,

(1) For future use case, i recommend u install git on prod. It will get rid
of all the install hassles. For each code update, all you have to do is:

git pull

(2) for the mean time, could u zip usbong code on prod and send me a copy?

I will identify if something is not up to date.

Have a good sunday ahead.

**Comment by masarapmabuhay** [comment]
Hi Zent,
Thank you for the reply.

I was able to resolve the issue.

As it turned out, when I do a `select now()` on mysql on prod, I am given an incorrect datetime stamp.

I modified mysql's config file to make the update permanent, so that even after doing a system restart the configuration remains the same.

As advised at [Stackoverflow](https://stackoverflow.com/questions/19903610/how-to-set-permanent-default-timezone-in-mysql-server/19904846), here are the steps:
1) `sudo vi /etc/mysql/mysql.conf.d/mysqld.cnf`

2) Insert the following line:
--> `i`
default_time_zone='+08:00'
--> `:wq`

3) `sudo service mysql restart`

This case is now closed.

Thank you.

Peace,
Mike
